src/pages/Login/Login.js

```diff
@@ -0,0 +1,101 @@
+import React, { useReducer, useState } from 'react'
+import './Login.css'
+import { useAuth } from 'context'
+import { loginFormReducer } from 'reducer'
+import axios from 'axios'
+import { useNavigate, Link } from 'react-router-dom'
+const Login = () => {
+    const navigate = useNavigate()
+    const { authDispatch } = useAuth();
+
+    const [{ email, password }, loginDispatch] = useReducer(loginFormReducer, { email: "", password: "" })
+
+
+    const [togglePassword, setTogglePassword] = useState(false);
+    const showPassword = togglePassword ? "visibility_off" : "visibility";
+    const passwordVisibilityHandler = () => {
+        setTogglePassword(!togglePassword)
+    }
+
+    const testHandler = () => {
+        loginDispatch({ type: "SET_EMAIL", payload: "yashpatel@gmail.com" })
+        loginDispatch({ type: "SET_PASSWORD", payload: "yashpatel" })
+
+    }
+
+    const submitHandler = async (e, email, password) => {
+        e.preventDefault();
+        try {
+
+            const response = await axios.post("api/auth/login", { email, password });
+
+
+            localStorage.setItem("token", response.data.encodedToken)
+            localStorage.setItem('userData', JSON.stringify(response.data.foundUser));
+            authDispatch({ type: "USER_LOGIN" })
+            authDispatch({ type: "TOKEN_ADD", payload: response.data.encodedToken })
+            authDispatch({ type: "USER_DATA_ADD", payload: response.data.foundUser })
+        } catch (error) {
+            console.log(error);
+        } finally {
+            navigate("/")
```

Comment:
If you use finally, it will run even if there is an error. That would not be the right ux

Reply:
yes, haven't done error handling yet will update this.